tardis/datatable.py

```diff
@@ -0,0 +1,314 @@
+         ----------
+         df = pd.DataFrame( 
+                data = pd.np.random.randint(0, 100, (10, 5)) , 
+                columns = list('ABCED') )
+         
+         datatable = DataTable (df, units=[u.meter, u.second, u.kg, u.meter/u.second, u.Ohm]) 
+         
+         
+    Example to set units afterwards
+         ----------
+         datatable.units = pd.Series(['m', 's', 't', 'm/s', 'kg'])
+         
+         """
+
+    _metadata = ['units']
+
+
+    @property
+    def _constructor(self):
+        return DataTable
+
+
+    def __init__(self, *args, **kwargs):
+        self.units = pd.Series(kwargs.pop("units", None))
+        super().__init__(*args, **kwargs)
+
+
+    def copy(self):
+        """
+        Creates a copy of the given class
+        Returns
+        --------
+        Object of DataTable class
+        
+        """
+        x = DataTable(self)
+        x.units = self.units
+        return x
+
+
+    def get_unit(self, attr):
+
+        """
+        Return unit of a particular attribute.
+        Parameters
+        ----------
+        attr: string or int
+            name of the attribute
+            
+        Returns
+        ----------
+        unit of the attribute
+        
+        """
+        ind = list (self.columns).index(attr)
+        if len((self.units)) > max (0 ,ind) :
+            return self.units[ind]
+        else :
+            print( "The unit for the following attribute is not set." )
+            
+    
+    def scalar_mul(self, const, unit, attr):
+        
+        """
+        Multiplies a constant having unit to entire datatable
+        
+        Parameters
+        ----------
+        const: int
+            value of the scalar
+            
+        unit: astropy
+            unit of the scalar
+        
+        attr: array of string or int
+            list of columns to which scalar is multiplied
+        
+        """        
+        new_units = []
+        for i in range(len(self.units)):
+            if( list(self.columns) [i] in attr):
+                new_units.append(self.units[i]*unit)
+            else:
+                new_units.append(self.units[i])
+        
+        for i in attr:
+            if(i not in self.columns):
+                print("Wrong Attributes entered")
+                break
+            self[i] = const*self[i]
+            
+        self.units = pd.Series(new_units)   
+       
+    
+    def dot(self, series, unit):
+        
+        """
+        Multiplies a constant having unit to some columns of datatable
+        
+        Parameters
+        ----------
+        series: Series or array
+            values of the series
+            
+        unit: astropy.units
+            unit of the scalar
+
+        Returns
+        ----------
+        object of class DataTable
+        
+        """
+        new_units = []
+        for i in self.units:
+            new_units.append(i*unit)
+
+        dataframe = super(DataTable,self).dot(series)
```

Comment:
This is exactly what I wanted.

Reply:
Thanks @marxwillia . Finally, I was able to crack. Thanks for making me think in this direction. Really learned a cool stuff :)